### PR TITLE
CEDS-2406 EXS Declaration question with logic around parties pages

### PR DESCRIPTION
**Pull request by PatrykRudnicki**

### CODE REVIEW

File: app/controllers/declaration/CarrierDetailsController.scala [hunk 1]
```diff
@@ -49,8 +51,8 @@ class CarrierDetailsController @Inject()(
   def displayPage(mode: Mode): Action[AnyContent] =
     (authenticate andThen journeyType(validTypes)) { implicit request =>
       request.cacheModel.parties.carrierDetails match {
-        case Some(data) => Ok(carrierDetailsPage(mode, navigationForm, form().fill(data)))
-        case _          => Ok(carrierDetailsPage(mode, navigationForm, form()))
+        case Some(data) => Ok(carrierDetailsPage(mode, navigationPage, form().fill(data)))
```
Comment (SteveSugden) [inline app/controllers/declaration/CarrierDetailsController.scala:54]: Are you getting together with @MaciejRewera to get rid of this navigationPage nonsense?

Comment (PatrykRudnicki) [reply app/controllers/declaration/CarrierDetailsController.scala:54]: Yeah, I talked with him, but this PR will go first so I don't have the Navigator logic

File: app/models/requests/JourneyRequest.scala [hunk 2]
```diff
@@ -25,4 +25,6 @@ class JourneyRequest[+A](val authenticatedRequest: AuthenticatedRequest[A], val
   val sourceDecId: Option[String] = cacheModel.sourceId
   def isType(`type`: DeclarationType*): Boolean = `type`.contains(declarationType)
   def eori: String = authenticatedRequest.user.eori
+
+  def isDeclarantExporter: Boolean = cacheModel.parties.declarantIsExporter.exists(_.isExporter)
```
Comment (MaciejRewera) [inline app/models/requests/JourneyRequest.scala:29]: I don't think this is appropriate place for this method. I'd rather move it to `ExportsDeclaration`

File: test/models/requests/JourneyRequestSpec.scala [hunk 3]
```diff
@@ -52,5 +52,35 @@ class JourneyRequestSpec extends UnitSpec with ExportsDeclarationBuilder with Mo
...
+
+      "user didn't answer on this question" in {
+
+        val declaration = aDeclaration(withType(DeclarationType.OCCASIONAL))
+        val authenticatedRequest = getAuthenticatedRequest()
+        val request = new JourneyRequest(authenticatedRequest, declaration)
+
+        request.isDeclarantExporter mustBe false
+      }
+    }
```
Comment (MaciejRewera) [inline test/models/requests/JourneyRequestSpec.scala:84]: Similarly, I don't agree this is the place for this method.

File: test/unit/forms/declaration/ExporterDetailsSpec.scala [hunk 4]
```diff
@@ -25,12 +25,6 @@ class ExporterDetailsSpec extends WordSpec with MustMatchers with LightFormMatch
 
   import forms.declaration.ExporterDetailsSpec._
 
-  "Exporter Details" should {
-    "have correct id" in {
-      ExporterDetails.id mustEqual "ExporterDetails"
-    }
-  }
-
```
Comment (MaciejRewera) [inline test/unit/forms/declaration/ExporterDetailsSpec.scala:33]: Why is this removed if in IsExcSpec you are testing the same thing?

File: test/unit/forms/declaration/IsExsSpec.scala [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+        IsExs.form.fillAndValidate(IsExs("Yes")).errors mustBe empty
+      }
+
+      "the answer is No" in {
+
+        IsExs.form.fillAndValidate(IsExs("No")).errors mustBe empty
+      }
+    }
+  }
```
Comment (MaciejRewera) [inline test/unit/forms/declaration/IsExsSpec.scala:48]: This curly brace should be moved to the end of the file. Test names are incorrect now.

File: test/unit/forms/declaration/ExporterDetailsSpec.scala (hunk 4 above)
Comment (PatrykRudnicki) [reply test/unit/forms/declaration/ExporterDetailsSpec.scala:33]: We're not using the ExporterDetails Id anywhere. Removed ID and test for it

Comment (PatrykRudnicki) [reply test/unit/forms/declaration/ExporterDetailsSpec.scala:33]: IsExs tests key, not page ID because we're not using page id anymore

File: app/models/requests/JourneyRequest.scala (hunk 2 above)
Comment (PatrykRudnicki) [reply app/models/requests/JourneyRequest.scala]: Good point. Moved

File: test/models/requests/JourneyRequestSpec.scala (hunk 3 above)
Comment (PatrykRudnicki) [reply test/models/requests/JourneyRequestSpec.scala]: Moved

File: test/unit/forms/declaration/IsExsSpec.scala (hunk 5 above)
Comment (PatrykRudnicki) [reply test/unit/forms/declaration/IsExsSpec.scala]: Moved

File: test/views/declaration/CarrierDetailsViewSpec.scala [hunk 6]
```diff
@@ -146,6 +147,40 @@ class CarrierDetailsViewSpec extends UnitViewSpec with CommonMessages with Stubs
       }
     }
 
+    onClearance { implicit request =>
+      "display 'Back' button that links to 'Is Exs' page" in {
```
Comment (MaciejRewera) [inline test/views/declaration/CarrierDetailsViewSpec.scala:151]: 👍

File: test/views/declaration/ConsigneeDetailsViewSpec.scala [hunk 7]
```diff
@@ -305,6 +306,16 @@ class ConsigneeDetailsViewSpec extends UnitViewSpec with CommonMessages with Stu
...
+
+        val backButton = createView(navigationForm = ConsignorEoriNumber).getElementById("back-link")
+
+        backButton.text() mustBe messages(backCaption)
+        backButton.attr("href") mustBe routes.IsExsController.displayPage().url
+      }
+    }
+
     onJourney(DeclarationType.SUPPLEMENTARY) { implicit request =>
       "display 'Back' button that links to 'Representative Details' page" in {
```
Comment (MaciejRewera) [inline test/views/declaration/ConsigneeDetailsViewSpec.scala:320]: This case should be tested for all journeys. But I can fix it during my rebase if you want.

File: test/views/declaration/RepresentativeDetailsAgentViewSpec.scala [hunk 8]
```diff
@@ -86,8 +86,6 @@ class RepresentativeDetailsAgentViewSpec extends UnitViewSpec with ExportsTestDa
       }
 
       onJourney(DeclarationType.CLEARANCE) { implicit request =>
-        val view = createView()
-
         "display 'Back' button that links to 'Consignor Details' page" in {
```
Comment (MaciejRewera) [inline test/views/declaration/RepresentativeDetailsAgentViewSpec.scala:89]: Similar here. This case should be tested for all journeys.

File: test/views/declaration/ConsigneeDetailsViewSpec.scala (hunk 7 above)
Comment (PatrykRudnicki) [reply test/views/declaration/ConsigneeDetailsViewSpec.scala:320]: Yeah, if you can it will be great
